### PR TITLE
Move some codepaths away from MixedRealityToolkitFiles

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview

MixedRealityToolkitFiles has always had some issues around Assets vs Packages, relative vs absolute, and some less-standard forms of referencing MRTK2.

This PR moves away from most of the more problematic cases for in-MRTK code to use Unity's `AssetDatabase` APIs instead.

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4904

### CODE REVIEW

**Review by davidkline-ms — APPROVED** [review]

File: Assets/MRTK/Core/Utilities/Editor/EditorProjectUtilities.cs [hunk 1]
```diff
@@ -85,8 +85,7 @@ public static bool FindRelativeDirectory(string packageDirectory, out string pat
         /// </param>
```
Comment (K780420) [inline Assets/MRTK/Core/Utilities/Editor/EditorProjectUtilities.cs:85]: image

**Review by K780420 — COMMENTED** [review]
image

**Review by K780420 — COMMENTED** [review]
image

**Review by K780420 — COMMENTED** [review]
image

**Pull request merged**